src/dune/ml_template.mll

```diff
@@ -0,0 +1,64 @@
+{
```

Comment:
A new templating engine? Surely we've got enough existing ones between expect tests, cinaps etc?
@diml, do you have a recommendation of a simple templating mechanism for .ml files?

Reply:
Cinaps would work if it was possible to use it in dune. It was discussed in slack why dune's `String_with_vars.t` doesn't work well. The gist is that we need something that will be a valid template and a valid OCaml source

Reply:
I guess people will excuse me if I don't read it very carefully...

Reply:
Yes, no worries. It's quite good to have a public record of all this discussion.

Reply:
just a thought though: given that it is not a fully fledged templating engine, I'd rather we keep it hidden in the module where we use it. Personally, I'd just use something like this which is less generic but does the job with less code:

```ocaml
let replace_vars s repl =
  let rec process lines =
    match line with
    | [] -> []
    | line :: lines -> (
      match String.trim line with
      | "(*BEGIN_VARS*)" ->
        line :: repl
        :: List.drop_while lines ~f:(fun line ->
               String.trim line <> "(*END_VARS*)")
      | _ -> line :: process lines )
  in
  String.split_lines s |> process |> String.concat ~sep:"\n"
```

Alternatively, @rgrinberg if you have other use cases in mind for this, the current state is fine but it needs to be documented. i.e. one should be able to only look at `ml_template.mli` and understand what this module is about and how it can be used. That includes describing the syntax of templates for instance.

Reply:
@diml I've moved it to a private module. I've simplified the implementation, but still kept it whitespace agnostic inside the comments. I think it's too easy to make a mistake with whitespace otherwise.